### PR TITLE
correct spelling, minor word ordering

### PR DESCRIPTION
**Pull request by kbhawkey**
Hopefully, the descriptions of the transformer concepts have not been changed by my clean-up.
This is my first pass at understanding the concepts.

### CODE REVIEW

File: examples/transformerconfigs/README.md [hunk 1]
```diff
@@ -44,26 +51,30 @@ commonLabels:
 - path: spec/selector/matchLabels
   create: true
   kind: Deployment
-  (etc.)  
+# add additional paths to resource fields
```
Comment (Liujingfang1) [inline examples/transformerconfigs/README.md]: The comment `# add additional paths to resource fields` is misleading. How about change it to `truncated`.

File: examples/transformerconfigs/README.md [hunk 2]
```diff
@@ -44,26 +51,30 @@ commonLabels:
...
   version: v1
   path: spec/volumes/configMap/name
 - kind: Deployment
   path: spec/template/spec/volumes/configMap/name
 - kind: Job
   path: spec/template/spec/volumes/configMap/name
-  (etc.)
-```
-Name reference transformer configuration contains a list of such namebackreferences for ConfigMap, Secret, Service, Role, ServiceAccount and so on.
+# add additional paths to resource fields
```
Comment (Liujingfang1) [inline examples/transformerconfigs/README.md]: Same as above. How about `truncated` or not adding the comment at all?

File: examples/transformerconfigs/README.md [hunk 3]
```diff
@@ -84,12 +95,14 @@ nameReference:
   - path: spec/containers/env/valueFrom/secretKeyRef/name
     version: v1
     kind: Pod
-    (etc.)    
+  # add additional paths to resource fields
 ```
 
-## customizing transformer configurations
+## Customizing transformer configurations
+
```
Comment (Liujingfang1) [inline examples/transformerconfigs/README.md]: Suggest to change it to following paragraphs.

Kustomize has a default set of transformer configurations. You can view the default transformer configurations by saving them
to a local directory by calling `kustomize config save -d`. 

Kustomize also supports adding extra transformer configurations by adding configuration files in the kustomization.yaml file. This tutorial shows how to customize those configurations to:

File: examples/transformerconfigs/README.md [hunk 4]
```diff
@@ -1,37 +1,44 @@
...
-If `create` is set to true, it indicates the transformer to create the path if it is not found in the resources. This is most useful for label and annotation transformers, where the path for labels or annotations may not be set before the transformation.
 
-## prefix/suffix transformer
-Name prefix suffix transformer adds prefix and suffix to the `metadata/name` field for all resources with following configuration:
-```
+If `create` is set to `true`, the transformer creates the path to the field in the resource if the path is not already found. This is most useful for label and annotation transformers, where the path for labels or annotations may not be set before the transformation.
+
+## Prefix/suffix transformer
+
+The prefix/suffix transformer adds a prefix/suffix to the `metadata/name` field for all resources. Here is an example of a prefix transformer configuration:
```
[the remark this answers is not in the document]
Comment (kbhawkey) [reply examples/transformerconfigs/README.md]: yes, will change

File: examples/transformerconfigs/README.md (hunk 2 above)
Comment (kbhawkey) [reply examples/transformerconfigs/README.md]: I guess no comment will do.

File: examples/transformerconfigs/README.md [hunk 5]
```diff
@@ -1,37 +1,44 @@
...
-Each transformer takes a list of resources and modifies certain fields. The modification is based on the transformer's rule.
-The fields to update is the transformer's configuration, which is a list of filedspec that can be represented in YAML format.
+Each transformer takes a list of resources and modifies certain fields in the resource based upon the transformer's configuration. A transformer's configuration is a list of `fieldSpec`, represented in YAML format.
 
-## fieldSpec
-FieldSpec is a type to represent a path to a field in one kind of resources. It has following format
-```
+## FieldSpec
+
+FieldSpec is a type that represents a path to a field in a resource such as `Job`. Here is the `fieldSpec` format:
```
[the remark this answers is not in the document]
Comment (kbhawkey) [reply examples/transformerconfigs/README.md]: OK

File: examples/transformerconfigs/README.md (hunk 1 above)
Comment (kbhawkey) [reply examples/transformerconfigs/README.md]: I'll take out the comment. Add back if needed.

File: examples/transformerconfigs/README.md [hunk 6]
```diff
@@ -1,37 +1,44 @@
 # Transformer Configurations
 
-Kustomize computes the resources by applying a series of transformers:
+Kustomize computes the configuration of a resource by applying a series of transformers:
```
Comment (monopole) [inline examples/transformerconfigs/README.md]:
```
Kustomize creates new resources by applying
a series of transformations to an original
set of resources.
```

File: examples/transformerconfigs/README.md [hunk 7]
```diff
@@ -1,37 +1,44 @@
...
 - namespace transformer
 - prefix/suffix transformer
 - label transformer
 - annotation transformer
 - name reference transformer
 - variable reference transformer
 
-Each transformer takes a list of resources and modifies certain fields. The modification is based on the transformer's rule.
-The fields to update is the transformer's configuration, which is a list of filedspec that can be represented in YAML format.
+Each transformer takes a list of resources and modifies certain fields in the resource based upon the transformer's configuration. A transformer's configuration is a list of `fieldSpec`, represented in YAML format.
```
Comment (monopole) [inline examples/transformerconfigs/README.md]:
```
A `fieldSpec` list in a transformer's configuration
determines which resource types and which fields
within those types the transformer can modify.
```

File: examples/transformerconfigs/README.md [hunk 8]
```diff
@@ -1,37 +1,44 @@
...
 
-## fieldSpec
-FieldSpec is a type to represent a path to a field in one kind of resources. It has following format
-```
+## FieldSpec
+
+FieldSpec is a type that represents a path to a field in one kind of resource, such as `Job`.
```
Comment (monopole) [inline examples/transformerconfigs/README.md]: No need to mention Job here - it just confuses me.

File: examples/transformerconfigs/README.md (hunk 6 above)
Comment (kbhawkey) [reply examples/transformerconfigs/README.md]: OK. How to differentiate between resources defined in manifests and live resources/object?

File: examples/transformerconfigs/README.md (hunk 7 above)
Comment (kbhawkey) [reply examples/transformerconfigs/README.md]: OK, will update.

File: examples/transformerconfigs/README.md (hunk 8 above)
Comment (kbhawkey) [reply examples/transformerconfigs/README.md]: OK

File: examples/transformerconfigs/README.md (hunk 6 above)
Comment (kbhawkey) [reply examples/transformerconfigs/README.md]: What about this possible change to below:

Kustomize provides the following default transformers:

- namespace
- prefix/suffix
- label
- annotation
- name reference
- variable reference

**Comment by monopole** [comment]
/lgtm
/approve
thanks @kbhawkey !  :)

**Pull request merged**